paleofetch.c

```diff
@@ -671,6 +683,7 @@ int main(int argc, char *argv[]) {
     }
 
     int offset = 0;
+    char *LOGO = *ARCH_LOGO;
```

Comment:
This is the culprit.
ARCH_LOGO is an array of char pointers.
You create a new LOGO variable of type char pointer to which you assign the dereferenced value of ARCH_LOGO.
Unfortunately this does not throw a compiler error, because `*ARCH_LOGO` is also of type char pointer.
Why though?
The dereference operator in C ( `*` ) means: treat ARCH_LOGO as an memory address and return the value of whatever is written at this address.
ARCH_LOGO is an array of char pointers. When using the name of an array in C code, it will give you the address of the first element in the array, which when dereferenced is a char pointer.
But this is not what you want to accomplish. What you got was a variable, that points only to the first line of the logo.
If you want to point to ARCH_LOGO itself, you would need a pointer to a char pointer, which would look like this:
`char **LOGO = ARCH_LOGO;`
But this does not work with the existing code directly, because the COUNT macro does not work the same way with char pointer pointers, as it does with char pointer arrays. Also I don't think that this can work with logos that have different line lengths, because when you access different indices of the logo, C needs to know where exactly to look for the start of the string. When you only handle a pointer to an array, there is no way to know this during compile time.

Reply:
I am new to C (I'm coming from python)

Reply:
So how should this issue be tackled?

Reply:
No worries, everyone has to start somewhere. This is already quite advanced C stuff and I would guess that most people who once learned C somewhere don't really understand what's going on here (I also had to think quite hard about this).
I also write a lot of stuff in Python. I like it, because it's simple, effective and has it's own unique style.

Reply:
@dwzg since python is easy to code for, and cython exists. Why don't people just code in python and convert it to C via cython, then compile it? Is there a performance punishment? I know python is slow compared to C, but when it's compiled, wouldn't it be the same speed? I feel like it would make a lot of stuff easier. Why isn't this an approach that is more widely used? 
Just wondering.

Reply:
I've done some testing, a neofetch-like python program that I made (not fully done, but gives a good idea for a full port to python) performs about 10x slower with cython, but pure-python performance varies, but always slower than paleofetch. So python isn't a good idea.